skopt/acquisition.py

```diff
@@ -264,3 +267,119 @@ def gaussian_ei(X, model, y_opt=0.0, xi=0.01, return_grad=False):
+        It should have a ``return_std`` parameter that returns the standard
+        deviation.
+
+    * `space` [Space instance]:
+        This is required to sample representer points.
+
+    * `n_rep_points` [int, default 50]:
+        Number of represter points.
+
+    * `n_trial_points` [int, default 200]:
+        Number of trial points sampled uniformly for each representer point
+        candidate.
+
+    * `n_samples` [int, default 500]:
+        Number of samples to sample from the GP posterior to approximate p_min
+
+    * `random_state` [int, RandomState instance, or None (default)]:
+        For reproducible results.
+    """
+    rng = np.random.RandomState(random_state)
+    n_features = X.shape[1]
+
+    # Step 1: Sample `n_representer_points` from the given nonmeasure.
+    # XXX: Use MCMC sampling to sample from the quadrature measure.
+    n_total = n_rep_points*n_trial_points
+    rep_cands = space.transform(space.rvs(n_samples=n_total, random_state=rng))
+    rep_points = np.zeros((n_rep_points, n_features))
+
+    if measure == "LCB":
+        quad_func = gaussian_lcb
+
+    elif measure in ["EI", "PI"]:
+        if measure == "EI":
+            quad_func = lambda X, model: -gaussian_ei(X, model)
+        else:
+            quad_func = lambda X, model: -gaussian_pi(X, model)
+
+    for i, startptr in enumerate(range(0, X.shape[0], n_total)):
+        rep_curr = rep_cands[startptr: startptr+n_trial_points]
+        argmin = np.argmin(quad_func(rep_curr, model))
+        rep_points[i] = rep_curr[argmin]
+
+    # # Step 2: Calculate p_min for each of these represnter points using
+    # # Monte-Carlo approximation.
+    # sampled_f = model.sample_y(
+    #     X_candidates, n_samples=n_samples, random_state=rng)
+    # p_min = (np.bincount(np.argmax(sampled_f, axis=0)) + 1) / (n_candidates + float(n_samples))
+    # base_entropy = entropy(p_min)
+
+    X_all = np.vstack((rep_points, X))
+    X_all_mean, X_all_cov = model.predict(X_all, return_cov=True)
+
+    # Store mu(X_i) and Sigma(X_i, X_j) of all the representer points based on
+    # previous evaluations.
+    init_cov = X_all_cov[:n_rep_points, :n_rep_points]
+    init_mean = X_all_mean[:n_rep_points]
+
+    # For each candidate point calculate change in p_min of all the represnter
+    # points and hence change in entropy (information gain)
+    inf_gain = np.zeros(X.shape[0])
+    for cand_ind in range(n_rep_points, n_rep_points + X.shape[0]):
+
+        # delta(cov(X_rep)) = \Sigma(X_rep, X_cand) \Sigma^-1(X_cand, X_cand) \Sigma(X_cand, X_rep)
+        # Note that \Sigma(X_cand, X_cand) is a scalar since we only require the next
+        # best evaluation.
+        cov_rep_X = X_all_cov[cand_ind][: n_rep_points]
+        cov_rep_X_row = np.reshape(cov_rep_X, (-1, 1))
+        cov_rep_X_col = np.reshape(cov_rep_X, (1, -1))
+        cov_cand_inv = 1.0 / X_all_cov[cand_ind, cand_ind]
+        cov_delta = cov_cand_inv * np.dot(cov_rep_X_row, cov_rep_X_col)
```

Comment:
This is a good question. According to the paper, it should be a positive sign. I believe I checked it and came to the conclusion that it should be a negative sign and this was missing in the paper, but I cannot say for sure. I should have added a comment, sorry. It seems that RoBO has also a negative sign (https://github.com/automl/RoBO/blob/master/robo/acquisition_functions/information_gain.py#L263). But please double check (if you factor this part out, you could write a unit test for it).